BlockchainSdk/Blockchains/Koinos/KoinosTransactionBuilder.swift

```diff
@@ -0,0 +1,151 @@
+        
+        guard let params = transaction.params as? KoinosTransactionParams else {
+            throw KoinosTransactionBuilderError.unableToParseParams
+        }
+        
+        let satoshi = pow(10, Blockchain.koinos(testnet: false).decimalCount)
+        let amountSatoshi = (amount * satoshi).uint64Value
+        
+        let manaLimit = params.manaLimit
+        let manaLimitSatoshi = (manaLimit * satoshi).uint64Value
+        
+        let nextNonce = currentNonce.nonce + 1
+
+        let operation = try Koinos_Protocol_operation.with {
+            $0.callContract = try Koinos_Protocol_call_contract_operation.with {
+                $0.contractID = contractId.base58DecodedData
+                $0.entryPoint = transferEntryPoint
+                $0.args = try Koinos_Contracts_Token_transfer_arguments.with {
+                    $0.from = from.base58DecodedData
+                    $0.to = to.base58DecodedData
+                    $0.value = amountSatoshi
+                }
+                .serializedData()
+            }
+        }
+        
+        let operationSha256 = try operation.serializedData().getSha256()
+        let operationMerkleRoot = Data([18, 32] + operationSha256.bytes)
+        let encodedNextNonce = try Koinos_Chain_value_type.with {
+            $0.uint64Value = nextNonce
+        }
+        .serializedData()
+        
+        let header = Koinos_Protocol_transaction_header.with {
+            $0.chainID = chainId.base64UrlDecodedData()!
+            $0.rcLimit = manaLimitSatoshi
+            $0.nonce = encodedNextNonce
+            $0.operationMerkleRoot = operationMerkleRoot
+            $0.payer = from.base58DecodedData
+        }
+        
+        let hashToSign = try header.serializedData().getSha256()
+        let transactionId = "\(transactionIdPrefix)\(hashToSign.hexString.lowercased())"
+        
+        let transactionToSign = KoinosProtocol.Transaction(
+            header: KoinosProtocol.TransactionHeader(
+                chainId: chainId,
+                rcLimit: manaLimitSatoshi,
+                nonce: encodedNextNonce.base64UrlEncodedString(),
+                operationMerkleRoot: operationMerkleRoot.base64UrlEncodedString(),
+                payer: from,
+                payee: nil
+            ),
+            id: transactionId,
+            operations: [
+                KoinosProtocol.Operation(
+                    callContract: KoinosProtocol.CallContractOperation(
+                        contractIdBase58: contractId,
+                        entryPoint: Int(transferEntryPoint),
+                        argsBase64: operation.callContract.args.base64UrlEncodedString()
+                    )
+                )
+            ],
+            signatures: []
+        )
+        
+        return (transactionToSign, hashToSign)
+    }
+    
+    func buildForSend(transaction: KoinosProtocol.Transaction, normalizedSignature: Data) throws -> KoinosProtocol.Transaction  {
```

Comment:
думаю можно убрать, т.к. нигде ничего не выбрасывает ошибку
```suggestion
    func buildForSend(transaction: KoinosProtocol.Transaction, normalizedSignature: Data) -> KoinosProtocol.Transaction  {
```